src/Classes/Cli/HelpRenderer.php

```diff
@@ -0,0 +1,163 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of MMLC - ModifiedModuleLoaderClient.
+ *
+ * (c) Robin Wieschendorf <mail@robinwieschendorf.de>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace RobinTheHood\ModifiedModuleLoaderClient\Cli;
+
+class HelpRenderer
+{
+    private const INDENT = '  ';
+    private string $description = '';
+    private array $sections = [];
+
+    public function setDescription(string $description): void
+    {
+        $this->description = $description;
+    }
+
+    public function setUsage(string $command, string $description): void
+    {
+        $this->sections['usage'][$command] = $description;
+    }
+
+    public function addArgument(string $argument, string $description): void
+    {
+        $this->sections['arguments'][$argument] = $description;
+    }
+
+    public function addOption(string $short, string $long, string $description): void
+    {
+        $option = [
+            'short' => TextRenderer::color($short ? '-' . $short : '', TextRenderer::COLOR_GREEN),
+            'separator' => TextRenderer::color($short && $long ? ', ' : '', TextRenderer::COLOR_GREEN),
+            'long' => TextRenderer::color($long ? '--' . $long : '', TextRenderer::COLOR_GREEN),
+            'description' => $description,
+        ];
+
+        $this->sections['options'][] = $option;
+    }
+
+    public function render(): string
+    {
+        $render = '';
+
+        $render .= $this->renderDescription();
+        $render .= $this->renderUsage();
+        $render .= $this->renderArguments();
+        $render .= $this->renderOptions();
+
+        return $render;
+    }
+
+    private function renderDescription(): string
+    {
+        $description = \PHP_EOL;
+        $description .= TextRenderer::color('Description:', TextRenderer::COLOR_YELLOW) . \PHP_EOL;
+        $description .= self::INDENT . $this->description . \PHP_EOL;
+
+        return $description;
+    }
+
+    private function renderUsage(): string
+    {
+        if (empty($this->sections['usage'])) {
+            return '';
+        }
+
+        $items = \array_keys($this->sections['usage']);
+        $maxLength = TextRenderer::getMaxLength($items) + 1;
+
+        $usage = \PHP_EOL;
```

Comment:
Let the caller decide whether he wants an empty line or not. Do this in render()